course_catalog/etl/video.py

```diff
@@ -2,65 +2,63 @@
 
 from django.conf import settings
 from googleapiclient.discovery import build
-from django.conf import settings
+import urllib3
+import yaml
+import certifi
+
+CONFIG_FILE = 'https://raw.githubusercontent.com/mitodl/open-video-data/master/youtube.yaml'
+
+def youtube_client():
+    api_service_name = 'youtube'
+    api_version = 'v3'
+    developer_key = settings.YT_DEVELOPER_KEY
 
-class YouTubeApi:
-    """
-    Class interface to YouTube API calls
-    """
-
-    client = None
-
-    def __init__(self):
-        api_service_name = "youtube"
-        api_version = "v3"
-        developer_key = settings.YT_DEVELOPER_KEY
-
-        self.client = build(api_service_name, api_version, developerKey=developer_key)
-
-    def get_videos_for_channel_in_batches(self, channel_id, offered_by):
-        page_token = ""
-
-        while True:
-            request = self.client.search().list(
-                part="snippet",
-                channelId=channel_id,
-                maxResults=50,
-                pageToken=page_token
-            )
-
-            response = request.execute()
-            response["offered_by"] = offered_by
-
-         
-            for video in response['items']:
-                try:
-                    video_id = video['id']['videoId']
-                    video['url'] = 'https://www.youtube.com/watch?v=' + video_id
-                    video['snippet']['full_description'] = self.get_full_description(video_id)
-                except KeyError:
-                    print(video)
-
-            yield response
-
-            if 'nextPageToken' in response.keys():
-                page_token = response['nextPageToken']
-            else:
-                break
-
-    def get_full_description(self, video_id):
-        request = self.client.videos().list(
-            part="snippet",
-            id="1jrHVNJ4Zds"
+    return build(api_service_name, api_version, developerKey=developer_key)
+
+def get_videos_for_channel_in_batches(youtube, channel_id, offered_by):
+    page_token = ""
+
+    while True:
+        request = youtube.search().list(
+            channelId=channel_id,
+            maxResults=50,
+            pageToken=page_token,
+            part='snippet',
+            type='video'
         )
+
         response = request.execute()
+        descriptions = []
 
-        return response["items"][0]["snippet"]["description"]
+        for video in response['items']:
+            video_id = video['id']['videoId']
+            descriptions.append(get_full_description(youtube, video_id))
+
+        yield (offered_by, response, descriptions)
```

Comment:
I think you want to move this `yield` into the loop above and make it `yield (offered_by, video, get_full_description(youtube, video_id))`